packages/salesforcedx-vscode-core/src/commands/util/libraryCommandlet.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { ToolingRetrieveResult } from '@salesforce/source-deploy-retrieve/lib/tooling/retrieve';
+import { ProgressLocation, window } from 'vscode';
+import { channelService } from '../../channels';
+import { nls } from '../../messages';
+import { notificationService } from '../../notifications';
+import { TelemetryData, telemetryService } from '../../telemetry';
+import { OrgAuthInfo } from '../../util';
+import { CommandletExecutor } from './sfdxCommandlet';
+
+export abstract class LibraryCommandletExecutor<T>
+  implements CommandletExecutor<T> {
+  protected showChannelOutput = true;
+  protected orgConnection: Connection | undefined;
+  protected executionName: string = '';
+  protected startTime: [number, number] | undefined;
+  protected telemetryName: string | undefined;
+
+  public execute(response: ContinueResponse<T>): void {}
+
+  public async build(
+    execName: string,
+    telemetryLogName: string
+  ): Promise<void> {
```

Comment:
@brpowell this will change once we start working in having a Client in the library that shares the connection with the rest of the functionality.